common/formatting/align_test.cc

```diff
@@ -975,5 +976,351 @@ TEST_F(InferAmbiguousAlignIntentTest, DifferenceSufficientlySmall) {
             "threeeee    four\n");
 }
 
+// Creates columns tree with the same layout as the syntax tree
+template <const AlignmentColumnProperties& props>
+class SyntaxTreeColumnizer : public ColumnSchemaScanner {
+ public:
+  SyntaxTreeColumnizer() = default;
+
+  void Visit(const SyntaxTreeNode& node) override {
+    ColumnPositionTree* column;
+    if (!current_column_)
+      column = ReserveNewColumn(node, props);
+    else
+      column = ReserveNewColumn(*current_column_, node, props);
+
+    ValueSaver current_column_saver(&current_column_, column);
+    ColumnSchemaScanner::Visit(node);
+  }
+  void Visit(const SyntaxTreeLeaf& leaf) override {
+    if (!current_column_)
+      ReserveNewColumn(leaf, props);
+    else
+      ReserveNewColumn(*current_column_, leaf, props);
+  }
+
+ protected:
```

Comment:
`current_column_` being protected indicates that it is intended to be accessed in derived classes. Is this the case (I don't see any mention of current_column_ in this pull request in other classe though).

I strongly suggest to start out making everything private and once we do have to consider providing protected access, this should be a very conscious choice as it weakens the encapsulation.

Reply:
done